src/gui/MidiClipView.cpp

```diff
@@ -140,8 +142,61 @@ void MidiClipView::changeName()
 
 
 
+void MidiClipView::transposeSelection()
+{
+	const auto selection = getClickedClips();
+
+	// Calculate the key boundries for all clips
+	int highest = 0;
+	int lowest = NumKeys - 1;
+	for (ClipView* clipview: selection)
+	{
+		if (auto mcv = dynamic_cast<MidiClipView*>(clipview))
+		{
+			TimePos start, end;
+			int low, high;
+			if (mcv->getMidiClip()->notes().getBounds(start, end, low, high))
+			{
+				lowest = std::min(low, lowest);
+				highest = std::max(high, highest);
+			}
+		}
+	}
+
+	int minStep = 0 - lowest;
+	int maxStep = NumKeys - 1 - highest;
+	int semitones = QInputDialog::getInt(this, tr("Transpose"), tr("Semitones to transpose:"), 0, minStep, maxStep, 1);
```

Comment:
Of course, this now restricts to only do transposing that does not change the distance between the notes (i.e. does not bash all notes into one key). If that was wanted, ignore this comment.

Reply:
Oh right, just reading your comments that it was intended. Solving my own comment.